### DISCUSSION TITLE
Qual curso de ReactJS realizar?

### DISCUSSION THREAD

**Discussion by AnakinRibeiro** [frontendbr/forum#1455] [Discussão]
Fala pessoal. Atualmente estou formado em Análise e Desenvolvimento de Sistemas e trabalhando como TI na área de suporte a usuário e infra, porém estou decidido a me focar e ingressar na área de desenvolvimento web, especificamente no Frontend.

Possuo um bom nível de conhecimento em HTML, CSS, JS/ES6, PHP, OO e recentemente descobri o ReactJS e React Native e simplesmente me apaixonei pela lib (ou framework, como queiram chamar kkkk) e vi que o mercado para essas ferramentas é gigante.

Tendo dito isso, estou procurando um curso online para acompanhar e desenvolver minhas skills na ferramente e tentar carreira com ela. Comprei um curso da Cod3r na Udemy sobre React + Redux, porém não consegui terminar pois achei a didática do instrutor muito confusa.

Queria muito realizar o Bootcamp GoStack da Rocketseat que engloba NodeJs, ReactJS e ReactNative, por gostar muito do conteúdo dos caras no youtube e achar a didática do instrutor deles incrível , porém o preço é meio salgado (12x de R$165,00). Me interessei também pelos cursos da Alura porém vi muito gente falando que o conteúdo deles é muito desatualizado e fiquei meio com o pé atrás.

Pois bem, estou aceitando indicações de cursos sobre NodeJS, ReactJS e React Native, lembrando que podem ser em PT-BR ou Inglês. 

Dês de já, valeu galera!

**Comment by jair-anderson-souza** [comment]
Dá uma sacada na rocketseat @AnakinRibeiro

**Comment by AnakinRibeiro** [comment]
@jass2125 Já conheço eles. Devorei todos os cursos Starter deles que são free em uma semana. Essa passada que rolou a semana Omnistack deles tbm que foi absurda. Porém o curso pago deles que abriu a turma hoje ta meio fora do meu orçamento, então estou em busca de outra opção.

**Comment by rafaelnogueira1** [comment]
Cara te falar que o da RocketSeat vale a pena. O preço tá um tanto alto mas tu vai ter retorno disso saca. Eles tão bem voltados na ideia de montar um conteúdo que te garanta uma condição de disputar vaga até antes de tu terminar o curso. Vale a pena cara.

**Comment by devlucaslopes** [comment]
Eu fiz o bootcamp deles (turma 5) e valeu muito cara! 

Realmente é um investimento alto, mas vale cada centavo. Eu trabalhava com PHP e resolvi apostar no bootcamp para migrar de stack.

Hoje eu trabalho só com a stack JS e aprendi tudo pelo curso deles! Sem falar que você vai receber os materias atualizados pra sempre.

**Comment by AnakinRibeiro** [comment]
Entendo que é um investimento, mas infelizmente minha situação atual não permite, principalmente o limite do cartão de crédito kkkkkk, mas o próximo com certeza irei participar, porém até la quero continuar estudando.

**Comment by onlurking** [comment]
O Colt Steele recentemente lançou um curso de [React](https://www.udemy.com/modern-react-bootcamp/) na Udemy, eu não tive oportunidade  de ver o material ainda, mas ouvi falar muito do curso por aí.

**Comment by weider86-zz** [comment]
Fica ligado que esse é em portugês e quase sempre tem cupom:
https://www.udemy.com/curso-reactjs-ninja/

**Comment by ozeas** [comment]
> Fica ligado que esse é em portugês e quase sempre tem cupom:
> https://www.udemy.com/curso-reactjs-ninja/

Esse vale muito à pena. Inclusive está em promoção na udemy!!

**Comment by gabriel-hahn** [comment]
Referente ao curso do Daciuk, tem esse [site](https://queroser.ninja/promocoes) dele com os cupons de desconto. O curso que o @onlurking citou também parece ser uma boa, além dos cursos do Stephen Grider na Udemy.
Para alguns tópicos mais avançados, tem o curso do [Wes Bos](https://advancedreact.com/) com um desconto para brasileiros :smile: .
O curso da Rocketseat é interessante por causa dos patterns que eles utilizam, a gente acaba aprendendo a desenvolver utilizando React com ferramentas e padrões utilizados por uma grande parte das empresas, além dos desafios propostos ao longo do bootcamp.

**Comment by Roger-Melo** [comment]
Em PT-BR, recomendo fortemente o [React Ninja](https://queroser.ninja/promocoes). A didática do professor é animal e o curso está em promoção.

Qualquer dúvida sobre o conteúdo pode ser postada nas issues do repo do curso. Dá uma olhada em como o professor é prestativo com os alunos: https://github.com/da2k/curso-reactjs-ninja/issues/196 . E ele é assim com **todo mundo** =)

**Comment by romulo2735** [comment]
Uma coisa que é referente ao conteudo, que noções de skill em Ecma preciso ter para usar o React?

**Comment by weider86-zz** [comment]
> Uma coisa que é referente ao conteudo, que noções de skill em Ecma preciso ter para usar o React?

https://willianjusten.com.br/fundamentos-javascript-antes-de-aprender-react/

**Comment by mdandrade** [comment]
E ai, @AnakinRibeiro , como foi a jornada no React?

**Comment by ninetails** [comment]
Quando o budget tiver de boas, [Frontend Masters](https://frontendmasters.com/).

**Comment by renerlemes** [comment]
@AnakinRibeiro já conseguiu fazer o curso da Rocketseat?

**Comment by wilsonneto-swe** [comment]
Já fiz o curso da Rocketseat, do DevPleno e vários outros, mas os dois foram os melhores e que mais influenciaram em minha carreira... 

Duas indicações pra quem ta buscando cursos de react, a rocketseat vai abrir a turma semana que vem, a última turma saia a R$ 1690/vaga... Caro rs mas vale a pena, eu estou terminando a turma passada e é muito bom.

Bootcamp RocketSeat R$ 1980,00:
https://rocketseat.com.br/week-10

E para quem achou muito caro tem também o curso do dev pleno que fiz antes até do da rocketseat e ele me ajudou a conquistar o cargo de fullstack que tenho hoje, e é bem mais em conta.

Curso Dev Pleno que ta saindo R$ 697,00:
http://bit.ly/reactjs-dev-pleno

Mas claro, a documentação própria é a base, mas cursos são bem bons para praticar com projetos prontos como a semana agora da rocketseat.

Apenas indicando, caso estejam buscando...
Abraços

**Comment by renerlemes** [comment]
@wilsonneto-dev você sabe se muda muita coisa da versão 8 ou 9 para a 10 que vai lançar na segunda-feira do curso da Rocketseat Bootcamp?

**Comment by wilsonneto-swe** [comment]
Fala @renerlemes, cara para ser sincero n'ao sei kk Mas acredito que se tiver atualiza;'ao eles v'ao liberar na plataforma pra gente que é das passadas.

Abraços!

**Comment by carloshrf** [comment]
@AnakinRibeiro, idem, sou da área de suporte, redes, infra e por ai. Atualmente estou me focando para o desenvolvimento web e assim como você estou focando nesta área e já estou indo atrás do GoStack. Pelo que conheci da rocketseat pelo omnistack, sinceramente, tem uma didática interessante e bem envolvente, estou até empolgado para começar essa caminhada.

**Comment by renerlemes** [comment]
@carloshrf fala comigo pelo e-mail, no seu perfil não tem o seu!

**Comment by wilsonneto-swe** [comment]
Siim, isso ai, ambos os cursos são muito bons, tanto o goStack quanto o Dev Pleno, eu particularmente gostei muito de ambos, os caras tem uma didática excelente e projetos interessantes de se fazer.

Omnistack: [https://rocketseat.com.br/gostack/10/inscricao](https://rocketseat.com.br/gostack/10/inscricao)

Dev Pleno: [http://bit.ly/reactjs-dev-pleno](http://bit.ly/reactjs-dev-pleno)

**Comment by carloshrf** [comment]
@wilsonneto-dev Por esses e mais motivos, tenho minhas dúvidas quanto a stack mas espero poder resolver isso tudo no meio do processo, tenho ótimas expectativas!

**Comment by italosilva01-zz** [comment]
Vocês sabem algum livro sobre  ?

**Comment by kevinshibuya** [comment]
> Vocês sabem algum livro sobre ?

Melhor leitura que tu pode fazer sobre o tema:
https://pt-br.reactjs.org/docs/getting-started.html

**Comment by wilsonneto-swe** [comment]
Discordo um pouco do comentário acima, mas o apoio muito.

Leia bastante a documentação, pratique e estude muito e sempre.

Eu separaria em três rodas, uma dentro da outra...

O react é o framework, a roda mais superficial, angular, vue, svelte, knockout, backbone etc, são frameworks, tenha sempre em mente que esta é a roda que mais gira e com mais velocidade, e a que menos te fará um ótimo programador em essência, não foque 100% aqui.

A roda debaixo e que gira em uma velocidade menor (porém gira bastante também) são as linguagens em si, abaixo dos frameworks, é o JS, C#, Java, pyton, swift, dart, kotlin, C, C++, etc. É bom ter uma base boa nestes caras, mas muito boa mesmo, pois frameworks vão e vem em uma velocidade incrível, ontem jquery era o supra sumo, hoje galera até torce o nariz rs Porém se você tem uma base sólida em JS vc anda entre frameworks tranquilamente, em questão de um mês ou menos vc consegue trocar de react para angular se sua base for sólida, mas ainda não julgo este o ponto mais importante...

Enfim a última roda, o núcleo disso tudo, os conceitos de algorítimos, arquitetura, design patterns, padrões arquiteturais, boas práticas, orientação a objetos, programação procedural, programação funcional, DDD, SOLID, Clean Code, MVC, MVVM, arquiteturas de sistemas distribuídos, CQRS, micro serviços, arquitetura em camadas, TDD com testes unitários, complexidade de algorítimos... e todo o mundo que serve de base para todas as camadas acima, acredito que seu foco deve estar aqui. Recomento que suas leituras sejão focadas aqui, e que faça cursos focados nestes assuntos também.

Eu já programei alguns anos com PHP, atualmente atuo com C# / .Net Core e vários serviços com node, e uma coisa que nestes 10 anos eu aprendi é que se você domina apenas o framework, vc é refém dele e isso pode sair caro, se vc domina a linguagem mas não domina conceitos por trás da engenharia de software e padrões arquiteturais e de projetos, seus sistemas poderão ficar extremamente difíceis de escalar e de dar manutenção... Agora se você focar na base de tudo, você consegue mudar de frameworks e até de linguagens com segurança e confiança, sabendo que seus projetos vão continuar tendo qualidade, manutenabilidade e escalabilidade. Já passei por Laravel e Phalcom com PHP, além de PHP puro, .Net Core e .Net Framework com C#, Nest e Express com Node, e em qualquer um destes, o padrão MVC, os patterns, DDD, TDD estão e devem estar presentes, e estes padrões muitas vezes farão mais diferença em seu projeto do que a linguagem ou framework adotado.

Apenas minha opinião... espero que ajude.

E dito isto segue algumas indicações que me ajudaram demais na carreira:
- Clean Code -  Robert Cecil Martin
- Domain Driven Design - Eric Evans
- Micro Serviços prontos para produção - Susan Fowler
- Design Patterns - Gang of Four

Abraços!

**Comment by luizamendes** [comment]
https://www.udemy.com/course/react-the-complete-guide-incl-redux/ Esse curso é bem completo, instrutor é ótimo e tem vários outros cursos também.

**Comment by wilsonneto-swe** [comment]
Aproveitando a reabertura desta thread, há algumas semanas atrás fiz uma lista dos cursos de React e JS que fiz no começo e no decorrer da minha carreira, e que me ajudaram demais em questões profissionais e de confiança com a tecnologia.

Para quem quiser ver a lista e minha opinião sobre cada:
https://github.com/wilsonneto-dev/Melhores_Cursos_Online

Abraços!

--
Att., Wilson Neto
Linkedin: https://www.linkedin.com/in/wilsonnetobr/ =)

**Comment by danielnass** [comment]
Já fiz vários, da Udemy e também do Wesbos, mas até agora o mais legal foi o do Kent C Dodds chamado [EpicReact](https://epicreact.dev)
